### PR TITLE
Update JsonSerializer docs for clarity.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Also updated some nits in JsonValueKind, JsonDocumentOptions.

cc @steveharter, @carlossanlop, @mairaw, @Jozkee

### CODE REVIEW

File: xml/System.Text.Json/JsonSerializer.xml [hunk 1]
```diff
@@ -46,7 +46,15 @@
...
-        <exception cref="T:System.Text.Json.JsonException">The JSON is invalid, <paramref name="returnType" /> is not compatible with the JSON, or when there is remaining data in the Stream.</exception>
+        <exception cref="T:System.Text.Json.JsonException">The JSON is invalid.
+          
+-or-
+
+<typeparamref name="returnType" /> is not compatible with the JSON.
+
+-or-
+
+There is remaining data in the span beyond a single JSON value.</exception>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/JsonSerializer.xml]: It doesn't make sense to use "stream" for the `string`/`ROS<byte>` overloads.

File: xml/System.Text.Json/JsonSerializer.xml [hunk 2]
```diff
@@ -269,11 +293,11 @@ The <xref:System.Text.Json.JsonReaderOptions> used to create the instance of the
           
 -or-
 
-<typeparamref name="TValue" /> is not compatible with the JSON.
+<typeparamref name="returnType" /> is not compatible with the JSON.
 
 -or-
 
-There is remaining data in the stream.</exception>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/JsonSerializer.xml]: If there is data left in the Utf8JsonReader, this overload doesn't throw. And using stream was incorrect here.

File: xml/System.Text.Json/JsonDocumentOptions.xml [hunk 3]
```diff
@@ -81,7 +81,7 @@ By default, `AllowTrailingCommas` is set to `false`, and a <xref:System.Text.Jso
 
 ## Remarks
 
-By default, a <exception cref="T:System.Text.Json.JsonException> is thrown if a comment is encountered.
+By default, a <xref:System.Text.Json.JsonException> is thrown if a comment is encountered.
```
Comment (carlossanlop) [inline xml/System.Text.Json/JsonDocumentOptions.xml]: Can't believe we didn't catch this before. The build should've failed.

Comment (rpetrusha) [reply xml/System.Text.Json/JsonDocumentOptions.xml]: Yes. It's disturbing that it didn't.

**Comment by rpetrusha** [comment]
I've already reviewed, @carlossanlop. I'll merge now.

**Pull request merged**